Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -208,43 +276,58 @@ namespace ROS2
 
     AZ::Name ROS2FrameComponent::GetJointName() const
     {
-        return AZ::Name(ROS2Names::GetNamespacedName(GetNamespace(), m_jointNameString).c_str());
+        return AZ::Name(ROS2Names::GetNamespacedName(GetNamespace(), m_jointName).c_str());
     }
 
-    void ROS2FrameComponent::SetJointName(const AZStd::string& jointNameString)
+    void ROS2FrameComponent::SetJointName(const AZStd::string& jointName)
     {
-        m_jointNameString = jointNameString;
+        m_jointName = jointName;
     }
 
     void ROS2FrameComponent::Reflect(AZ::ReflectContext* context)
     {
-        NamespaceConfiguration::Reflect(context);
+        if (auto jsonContext = azrtti_cast<AZ::JsonRegistrationContext*>(context))
+        {
+            jsonContext->Serializer<JsonFrameComponentConfigSerializer>()->HandlesType<ROS2FrameComponent>();
+        }
+
+        ROS2FrameConfiguration::Reflect(context);
         if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
         {
             serialize->Class<ROS2FrameComponent, AZ::Component>()
                 ->Version(1)
-                ->Field("Namespace Configuration", &ROS2FrameComponent::m_namespaceConfiguration)
                 ->Field("Frame Name", &ROS2FrameComponent::m_frameName)
-                ->Field("Joint Name", &ROS2FrameComponent::m_jointNameString)
-                ->Field("Publish Transform", &ROS2FrameComponent::m_publishTransform);
+                ->Field("Joint Name", &ROS2FrameComponent::m_jointName)
+                ->Field("Publish Transform", &ROS2FrameComponent::m_publishTransform)
+                ->Field("Namespace Configuration", &ROS2FrameComponent::m_namespaceConfiguration);
 
             if (AZ::EditContext* ec = serialize->GetEditContext())
             {
-                ec->Class<ROS2FrameComponent>("ROS2 Frame", "[ROS2 Frame component]")
+                ec->Class<ROS2FrameComponent>("ROS2 Frame Game Component [depreciated]", "[ROS2 Frame component]")
```

Comment:
```suggestion
                ec->Class<ROS2FrameComponent>("ROS2 Frame Game Component (deprecated)", "ROS2 Frame component")
```

Reply:
Deprecated might be a bit confusing since this component is still used, created on game mode enter from its editor component.

The serialization of this component is deprecated since it is replaced by configuration. Maybe good to describe it somewhere.

Reply:
:+1: on more explanation here.